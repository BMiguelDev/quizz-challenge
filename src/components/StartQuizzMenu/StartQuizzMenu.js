import React from 'react'

import styles from './StartQuizzMenu.module.scss';


export default function StartQuizzMenu({ quizzCategoriesArray, quizzOptions, handleQuizzStart, handleQuizzOptionChange }) {
    return (
        <div className={styles.unstarted_quizz_container}>
            <h2 className={styles.unstarted_quizz_title}>Quizz Challenge</h2>
            <p className={styles.unstarted_quizz_text}>
                Choose random trivia questions and <br />
                answer as many as you can!
            </p>
            <form onSubmit={handleQuizzStart}>
                <div className={styles.unstarted_quizz_inputs_container}>
                    <label>
                        Category
                        <br />
                        <select onChange={handleQuizzOptionChange} name="category" value={quizzOptions.category}>
                            <option value="">Any</option>
                            {quizzCategoriesArray.map(category => <option key={category.id} value={category.id}>{category.name}</option>)}
                        </select>
                    </label>
                    <label>
                        Difficulty
                        <br />
                        <select onChange={handleQuizzOptionChange} name="difficulty" value={quizzOptions.difficulty}>
                            <option value="">Any</option>
                            <option value="easy">Easy</option>
                            <option value="medium">Medium</option>
                            <option value="hard">Hard</option>
                        </select>
                    </label>
                </div>
                <button type="submit" className={styles.unstarted_quizz_button}>Start Quizz</button>
            </form>
        </div>
    );
}
