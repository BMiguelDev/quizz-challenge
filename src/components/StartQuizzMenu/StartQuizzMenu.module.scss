// Unstarted quizz page content
.unstarted_quizz_container {
    padding: 13rem 12rem;
    width: 100%;
    height: 92%;
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    .unstarted_quizz_title {
        margin: 0;
        font-size: 2.5rem;
        color: var(--main-title-color);
        letter-spacing: .1rem;
        padding-bottom: 1.5rem;
        transition: all .2s ease-in-out;
    }

    .unstarted_quizz_text {
        margin: 0;
        font-size: 1.05rem;
        color: var(--main-text-color);
        padding-bottom: 2.2rem;
        transition: all .2s ease-in-out;
    }

    .unstarted_quizz_inputs_container {
        display: flex;
        gap: 1rem;
        padding-bottom: 2.5rem;

        label {
            color: var(--main-text-color);
            transition: all .2s ease-in-out;

            select {
                margin-top: .35rem;
                overflow: hidden;
                text-overflow: ellipsis;
                font-family: 'Roboto', sans-serif;
                padding: .5rem;
                border-radius: 1rem;
                color: var(--main-text-color);
                background-color: var(--form-select-container-color);
                transition: all .2s ease-in-out;

                option {
                    padding: 1rem;
                }
            }

            &:first-of-type select {
                width: 90%;
            }

            &:nth-of-type(2) {
                width: 35%;
            }
        }
    }

    .unstarted_quizz_button {
        font-family: 'Roboto', sans-serif;
        background-color: var(--button-bg-color);
        color: var(--button-color);
        font-size: 1.2rem;
        padding: 1rem 5rem;
        border-radius: .75rem;
        border: none;
        cursor: pointer;
        font-family: 'Roboto', sans-serif;
        transition: all .2s ease-in-out;
    }
}


@import './media';
