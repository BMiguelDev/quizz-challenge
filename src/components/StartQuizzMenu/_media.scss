// Media query for mobile content (portrait)
@media screen and (max-width: 450px) {

     // Unstarted quizz page content
     .unstarted_quizz_container {
        padding: 7rem 3rem;

        .unstarted_quizz_title {
            font-size: 3rem;
        }

        .unstarted_quizz_text {
            font-size: 1.1rem;
        }

        .unstarted_quizz_inputs_container {

            label {
                font-size: 1.1rem;

                &:nth-of-type(1) {
                    flex: 2;
                }

                &:nth-of-type(2) {
                    flex: 1;
                }

                select {
                    padding: .55rem;
                    font-size: .95rem;
                }

                &:first-of-type select {
                    width: 100%;
                }

                &:nth-of-type(2) {
                    width: 100%;
                }
            }
        }

        .unstarted_quizz_button {
            font-size: 1.2rem;
            padding: 1.25rem 5rem;
        }
    }
}


// Media query for mobile content (landscape)
@media screen and (max-height: 750px) and (min-width: 451px) and (max-width: 950px) {

      // Unstarted quizz page content
      .unstarted_quizz_container {
        padding: 3rem;

        .unstarted_quizz_title {
            font-size: 2.15rem;
            padding-bottom: .95rem;
        }

        .unstarted_quizz_text {
            font-size: .9rem;
            padding-bottom: 1.25rem;
        }

        .unstarted_quizz_inputs_container {
            padding-bottom: 1.35rem;

            label {
                font-size: 1rem;

                &:nth-of-type(1) {
                    flex: 2;
                }

                &:nth-of-type(2) {
                    flex: 1;
                }

                select {
                    padding: .3rem;
                    font-size: .75rem;
                }

                &:first-of-type select {
                    width: 100%;
                }

                &:nth-of-type(2) {
                    width: 100%;
                }
            }
        }

        .unstarted_quizz_button {
            font-size: 1rem;
            padding: .75rem 4rem;
        }
    }
}
