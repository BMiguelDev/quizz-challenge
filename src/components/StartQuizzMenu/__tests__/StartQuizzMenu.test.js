import { cleanup, render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';

import StartQuizzMenu from "../StartQuizzMenu";


afterEach(() => {
    cleanup();
})


const quizzCategoriesTest = [
    {
        id: 1,
        name: "Test1"
    },
    {
        id: 2,
        name: "Test2"
    }
];

const quizzOptionsTest = {
    category: "categoryTest1",
    difficulty: "difficultyTest1",
    urlUsed: "urlTest1"
}

const propFunction1 = jest.fn();
const propFunction2 = jest.fn();


test('component should always match snapshot', () => {
    //const tree = renderer.create(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} />).toJSON();
    const tree = renderer.create(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} handleQuizzStart={propFunction1} handleQuizzOptionChange={propFunction2} />).toJSON();

    expect(tree).toMatchSnapshot();
});


test('both select inputs should have empty value', () => {
    // Arrange
    render(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} handleQuizzStart={propFunction1} handleQuizzOptionChange={propFunction2} />)

    // Act
    const selectElements = screen.getAllByRole('combobox');
    const [selectCategoryElement, selectDifficultyElement] = selectElements;

    const defaultOptionElements = screen.getAllByRole('option', { name: /any/i });
    const [defaultOptionElementCategory, defaultOptionElementDifficulty] = defaultOptionElements;

    // Assert
    expect(selectCategoryElement).toHaveValue("");
    expect(selectDifficultyElement).toHaveValue("");

    expect(defaultOptionElementCategory.selected).toBeTruthy();
    expect(defaultOptionElementDifficulty.selected).toBeTruthy();
})

test('category select input should have correct number of options', () => {
    render(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} handleQuizzStart={propFunction1} handleQuizzOptionChange={propFunction2} />)

    const categorySelectElement = screen.getByRole("combobox", { name: /category/i });

    expect(categorySelectElement.childElementCount).toEqual(3);
})

test('difficulty select input should have correct number of options', () => {
    render(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} handleQuizzStart={propFunction1} handleQuizzOptionChange={propFunction2} />)

    const difficultySelectElement = screen.getByRole("combobox", { name: /difficulty/i });

    expect(difficultySelectElement.childElementCount).toEqual(4);
})

test('quizz start button should be enabled', () => {
    render(<StartQuizzMenu quizzCategoriesArray={quizzCategoriesTest} quizzOptions={quizzOptionsTest} handleQuizzStart={propFunction1} handleQuizzOptionChange={propFunction2} />)

    const buttonElement = screen.getByRole('button');
    
    expect(buttonElement).toBeEnabled();
})
