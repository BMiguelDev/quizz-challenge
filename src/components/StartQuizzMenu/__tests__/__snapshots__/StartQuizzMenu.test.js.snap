// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should always match snapshot 1`] = `
<div
  className="unstarted_quizz_container"
>
  <h2
    className="unstarted_quizz_title"
  >
    Quizz Challenge
  </h2>
  <p
    className="unstarted_quizz_text"
  >
    Choose random trivia questions and 
    <br />
    answer as many as you can!
  </p>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="unstarted_quizz_inputs_container"
    >
      <label>
        Category
        <br />
        <select
          name="category"
          onChange={[MockFunction]}
          value="categoryTest1"
        >
          <option
            value=""
          >
            Any
          </option>
          <option
            value={1}
          >
            Test1
          </option>
          <option
            value={2}
          >
            Test2
          </option>
        </select>
      </label>
      <label>
        Difficulty
        <br />
        <select
          name="difficulty"
          onChange={[MockFunction]}
          value="difficultyTest1"
        >
          <option
            value=""
          >
            Any
          </option>
          <option
            value="easy"
          >
            Easy
          </option>
          <option
            value="medium"
          >
            Medium
          </option>
          <option
            value="hard"
          >
            Hard
          </option>
        </select>
      </label>
    </div>
    <button
      className="unstarted_quizz_button"
      type="submit"
    >
      Start Quizz
    </button>
  </form>
</div>
`;
